src/pid/environ.rs

```diff
@@ -0,0 +1,78 @@
+//! Process initial environment from `/proc/[pid]/environ`.
+
+use std::ffi::{OsStr, OsString};
+use std::fs::File;
+use std::io::{Result, Read};
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+
+use parsers::map_result;
+use nom::IResult;
+
+use libc::pid_t;
+
+/// A list of environment variables and their values.
+pub type Environ = Vec<(OsString, OsString)>;
+
+/// Parses the provided buffer.
+fn parse_environ(input: &[u8]) -> IResult<&[u8], Environ> {
+    // Parse 'key=value" pair.
+    named!(
+        pair<&[u8], (&[u8], &[u8])>,
+        tuple!(take_until_and_consume!("="), take_until!("\0"))
```

Comment:
This is going to have different behavior that the standard library when the key contains a leading '=' character (https://github.com/rust-lang/rust/blob/1.19.0/src/libstd/sys/unix/os.rs#L405-L408).  Can this skip a character first to avoid this?

Reply:
Also, please add a test case for this.

Reply:
Nonetheless it appears to be a real issue.  On my up-to-date linux machine (kernel 4.11) the output of `env =FOO=BAR xxd /proc/self/environ` shows that leading `=` characters can appear.